channeldb/invoices.go

```diff
@@ -982,6 +988,15 @@ func serializeHtlcs(w io.Writer, htlcs map[CircuitKey]*InvoiceHTLC) error {
 			tlv.MakePrimitiveRecord(expiryHeightType, &htlc.Expiry),
 			tlv.MakePrimitiveRecord(stateType, &state),
 		)
+
+		customRecords, err := tlv.MapToRecords(htlc.CustomRecords)
+		if err != nil {
+			return err
+		}
+
+		records = append(records, customRecords...)
+
+		tlvStream, err := tlv.NewStream(records...)
```

Comment:
Question open as to whether we want to bump the db version, to prevent users running master from downgrading and loosing their custom records

Reply:
Discussed this in the context of another PR. We don't give this guarantee to users running master generally.